plasmapy/particles/tests/test_decorators.py

```diff
@@ -145,15 +156,16 @@ class ClassWithDecoratedMethods:
 
     @particle_input
     def method_decorated_with_no_parentheses(self, particle: ParticleLike) -> Particle:
-        return particle
+        # Because run-time logic is needed, mypy will assume that
```

Comment:
```suggestion
        # Because run-time logic is needed, mypy is unable to infer
        # that @particle_input is doing a type conversion here. We 
        # would need a dynamic type checker to address this case.
```